docs/en/styling-and-theming/supplemental.md

```diff
@@ -0,0 +1,642 @@
+
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+
+import * as css from '../styles/MyWidget.m.css';
+
+export default class MyTsxWidget extends WidgetBase {
+	protected render() {
+		return <div classes={[css.myWidgetClass, css.myWidgetExtraClass]}>Hello from a Dojo TSX widget!</div>;
+	}
+}
+```
+
+When inspecting the CSS classes of these sample widgets in a built application, they will not contain `myWidgetClass` and `myWidgetExtraClass`, but rather obfuscated CSS class names similar to `MyWidget-m__myWidgetClass__33zN8` and `MyWidget-m__myWidgetExtraClass___g3St`.
+
+These obfuscated class names are localized to `MyWidget` elements, and are determined by Dojo's CSS modules build process. With this mechanism it is possible for other widgets in the same application to also use the `myWidgetClass` class name with different styling rules, and not encounter any conflicts between each set of styles.
+
+**Warning:** The obfuscated CSS class names should be considered unreliable and may change with a new build of an application, so developers should not explicitly reference them (for examples if attempting to target an element from elsewhere in an application).
+
+## Abstracting and extending stylesheets
+
+### CSS custom properties
+
+Dojo allows use of CSSnext features such as [custom properties and `var()`](http://cssnext.io/features/#custom-properties-var) to help abstract and centralize common styling properties within an application.
+
+Rather than having to specify the same values for colors or fonts in every widget's CSS module, abstract custom properties can instead be referenced by name, with values then provided in a centralized CSS `:root` pseudo-class. This separation allows for much simpler maintenance of common styling concerns across an entire application.
+
+Note that the `:root` pseudo-class is global within a webpage, but through Dojo's use of CSS modules, `:root` properties could potentially be specified in many locations across an application. However, Dojo does not guarantee the order in which CSS modules are processed, so to ensure consistency of which properties appear in `:root`, it is recommended that applications use a single `:root` definition within a centralized `variables.css` file in the application's codebase. This centralized variables file is a regular CSS file (not a CSS module) and can be `@import`ed as such in any CSS modules that require custom property values.
+
+For example:
+
+> src/themes/variables.css
+
+```css
+:root {
+	/* different sets of custom properties can be used if an application supports more than one possible theme */
+	--light-background: lightgray;
+	--light-foreground: black;
+
+	--dark-background: black;
+	--dark-foreground: lightgray;
+
+	--font: monospace;
+	--padding: 32px;
+}
+```
+
+> src/themes/myDarkTheme/MyWidget.m.css
+
+```css
+@import '../variables.css';
+
+.root {
+	font-family: var(--font);
```

Comment:
I think we had an issue with using a var for `font-family` in IE very recently

Reply:
@rorticus @tomdye do you remember this? An issue I think was fixed in themes for this?

Reply:
Is it worth just removing `font-family` from the examples?

Reply:
yes